hkmc2/shared/src/main/scala/hkmc2/syntax/ParseRule.scala

```diff
@@ -272,6 +272,20 @@ class ParseRules(using State):
             End(N),
           )
         ) { (lhs, rhs) => TypeDef(Als, lhs, rhs, N) },
+    Kw(`pattern`):
```

Comment:
Couldn't we share the code for type synonyms? Place it in a common shared function.